Daemon: Update Health endpoints IP on localNode.

eloycoto opened the pull request:
On commit `db248247854f7a161bd9009e01fa39cdbc2bfcd4` the GetLocalNode
changed and the values of the Health endpoints were now no updated
correctly into the localnode.

This commit updates HealthIps for localnode.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5117)
<!-- Reviewable:end -->

tgraf commented on daemon/daemon.go:
```diff
@@ -1353,8 +1353,14 @@ func NewDaemon() (*Daemon, error) {
+		log.Fatal("Error while trying to get local node")
+	}
+	localNode.UpdateHealthAddresses(node.GetIPv4HealthIP(), node.GetIPv6HealthIP())
+	localNode.OnUpdate()
```
This needs a new function in `pkg/node/local_node.go`
```
func NotifyLocalNodeUpdated() error {
        return nodeStore.UpdateLocalKeySync(GetLocalNode())
}
```

which is then called instead of `localNode.OnUpdate()`

tgraf commented on daemon/daemon.go:
```diff
@@ -1353,8 +1353,14 @@ func NewDaemon() (*Daemon, error) {
+	if localNode == nil {
+		log.Fatal("Error while trying to get local node")
+	}
+	localNode.UpdateHealthAddresses(node.GetIPv4HealthIP(), node.GetIPv6HealthIP())
```
Instead of doing this, let's just modify `localNode` from `node.SetIPv4HealthIP()`.

tgraf commented on daemon/main.go:
```diff
@@ -769,11 +769,14 @@ func initEnv(cmd *cobra.Command) {
 func runCiliumHealthEndpoint(d *Daemon) error {
 	// PingEndpoint will always fail the first time (initialization).
 	if err := health.PingEndpoint(); err != nil {
+		localNode := node.GetLocalNode()
```
This introduces race conditions now as localNode is not using any mutex to protect its fields. I would keep the old way of retrieving the health IPs.

tgraf replied:
The reason for this is that we now access localNode in the context of go routines

eloycoto committed:
Daemon: Update Health endpoints IP on localNode.

On commit `db248247854f7a161bd9009e01fa39cdbc2bfcd4` the GetLocalNode
changed and the values of the Health endpoints were now no updated
correctly into the localnode.

This commit updates HealthIps for localnode.

Fix #5111

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
